client/state/routing/middleware.js

```diff
@@ -26,8 +31,9 @@ export const restoreLastLocation = () => {
 			( lastPath ) => {
 				if ( ! hasInitialized &&
 						lastPath && lastPath !== '/' &&
-						action.path === '/' &&
-						! isOutsideCalypso( lastPath ) ) {
+						action.path === '/' && Object.keys( action.query ).length === 0 &&
```

Comment:
One last nit to pick -- we should probably make sure `action.query` is defined to avoid a `TypeError`